runtime/src/bank_forks.rs

```diff
@@ -259,6 +259,13 @@ impl BankForks {
         bank
     }
 
+    pub fn prepare_to_drop(&mut self) {
```

Comment:
why not just do a custom drop?

Reply:
this is done: https://github.com/anza-xyz/agave/pull/5866#discussion_r2049115813